File: order-management-fe/src/services/OrderService.js
import axios from "axios";

const REST_API_BASE_URL = 'http://localhost:8080/api/orders';

export const listOrders = (page, size, sortBy, sortDir, airportCode)=>axios.get(REST_API_BASE_URL, {
    params: { page, size, sortBy, sortDir, airportCode }});
export const addOrder = (order)=>axios.post(REST_API_BASE_URL+'/add-order', order);    
export const editOrder = (id,order)=>axios.put(REST_API_BASE_URL+'/'+id, order);
export const getOrder = (id)=>axios.get(REST_API_BASE_URL+'/'+id);
export const deleteOrder = (id)=>axios.delete(REST_API_BASE_URL+'/'+id);

File: order-management-be/src/main/java/com/fuel/order/demo/service/OrderService.java
package com.fuel.order.demo.service;

import com.fuel.order.demo.dto.OrderDTO;
import com.fuel.order.demo.entity.Order;
import org.springframework.data.domain.Page;

import java.util.List;

public interface OrderService {

    public OrderDTO addOrder(OrderDTO orderDTO);
    public OrderDTO getOrder(Long id);
    public void updateOrder(OrderDTO orderDTO, Long id);
    public void deleteOrder(Long id);
    Page<OrderDTO> getOrders(int page, int size, String sortBy, String sortDir, String airportCode);
}
